plugin/session_view.py

```diff
@@ -36,7 +36,7 @@ def __init__(self, listener: AbstractViewListener, session: Session) -> None:
         self.progress = {}  # type: Dict[int, ViewProgressReporter]
         settings = self.view.settings()
         buffer_id = self.view.buffer_id()
-        key = (session.config.name, buffer_id)
+        key = (session.config.name, session.window.id(), buffer_id)
```

Comment:
what bad behavior can happen if we don't pass the window.id to the key?

I saw the commit description(https://github.com/sublimelsp/LSP/pull/1575/commits/349d76439acefd2d7585b75f8311d088337bd5cf), but I still don't understand why.

Reply:
Because of [this variable](https://github.com/sublimelsp/LSP/blob/c0ae90fb3cbcc6e07ba1ce2b76ba8aedc600df73/plugin/hover.py#L78), the buffers are kept alive until the next time the `run` method is invoked.

Now, when we reset the view event listener due to a moved window, we create a new SessionView. That SessionView then looks up its SessionBuffer in the weak map. It finds that SessionBuffer in the weak map, because it's still alive due to the hover command data, and the session name and buffer ID are the same. So we have to key on the window ID as well.